xcube/core/gen2/local/resamplert.py

```diff
@@ -19,19 +18,148 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
+import cftime
+import pandas as pd
 import xarray as xr
 
 from xcube.core.gridmapping import GridMapping
+from xcube.core.resampling import resample_in_time
+from xcube.core.resampling.temporal import adjust_metadata_and_chunking
+from xcube.util.assertions import assert_instance
 from .transformer import CubeTransformer
 from .transformer import TransformedCube
 from ..config import CubeConfig
+from ..error import CubeGeneratorError
 
 
 class CubeResamplerT(CubeTransformer):
 
+    def __init__(self,
+                 cube_config: CubeConfig):
```

Comment:
Please remove. I don't see a need to do this in the constructor.  `cube_config` is passed to `transform_cube()`.

Reply:
If I remove this from the constructor, the parameter 'time_range' will most likely have been consumed by the CubeOpener and be unavailable to the CubeResamplerT. In consequence, the resampler would have no information about the requested start and ending time. 
I see the following alternatives: 
(a) ignore the time range in the resampler
(b) introdcue a new parameter 'time_resample_time_range' so it would not be consumed earlier
(c) adapt the generator framework so that it does not remove parameters from a config

Reply:
> (a) ignore the time range in the resampler

I have now implemented this solution. The CubeConfig is not passed to the constructor anymore.
If transform_cube is called with a cubeconfig including time_range, it is used. Otherwise the time range is derived from the cube.